File: src/core/entities/message.entity.ts
import { BaseEntity } from "@entities/base.entity";
import { Column, Entity, JoinColumn, ManyToOne } from "typeorm";
import { UserEntity } from "@entities/user.entity";
import { ChannelEntity } from "@entities/channel.entity";

@Entity({ name: "messages" })
export class MessageEntity extends BaseEntity {
  @Column()
  public text: string;

  @ManyToOne(() => UserEntity, (user) => user.messages, { eager: true })
  @JoinColumn({ name: "author_id" })
  public author: UserEntity;

  @ManyToOne(() => ChannelEntity, (channel) => channel.messages)
  @JoinColumn({ name: "channel_id" })
  public channel: ChannelEntity;
}

File: SorokChatServer.Logic/Entities/MessageEntity.cs
using SorokChatServer.Postgres.Entities;

namespace SorokChatServer.Logic.Entities;

public class MessageEntity : BaseEntity
{
    public MessageEntity()
    {
    }

    public MessageEntity(long id, DateTime createdAt, DateTime updatedAt, string text, string mac, UserEntity author)
    {
        Id = id;
        CreatedAt = createdAt;
        UpdatedAt = updatedAt;
        Text = text;
        Mac = mac;
        Author = author;
    }

    public string Text { get; set; }

    public string Mac { get; set; }

    public UserEntity Author { get; set; }
}